untrusted/src/lib.rs

```diff
@@ -0,0 +1,157 @@
+//
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+
+use std::{ffi::CString, mem::MaybeUninit, os::raw::c_int};
+
+include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+#[derive(Default)]
+pub struct Enclave {
+    // The filename of the enclave
+    filename: CString,
+
+    // The enclave ID, assigned by the sgx interface
+    // Will be `None` when the enclave has not been created.
+    id: Option<sgx_enclave_id_t>,
+
+    // `true` if the enclave should be created in debug mode
+    debug: bool,
+}
+
+impl Enclave {
+    /// Returns an Enclave for the provided signed enclave.
+    ///
+    /// # Arguments
+    ///
+    /// * `filename` - The name of the enclave file.  This should be a signed
+    ///     enclave.
+    pub fn new(filename: &str) -> Enclave {
+        let filename = CString::new(filename).expect("Can't convert enclave filename to CString.");
+        Enclave {
+            filename,
+            ..Default::default()
+        }
+    }
+
+    /// Toggle debugging of the enclave on or off.  The default is off.
+    ///
+    /// # Arguments
+    ///
+    /// * `debug` - `true` to enable enclave debugging, `false` to disable it.
+    pub fn debug(&mut self, debug: bool) -> &mut Enclave {
+        self.debug = debug;
+        self
+    }
+
+    /// Get the ID for this instance.  The ID will not be valid to use in SGX
+    /// calls once this instance has dropped.
+    ///
+    /// The return value is intentionally a reference to a `sgx_enclave_id_t`.
+    /// This allows consumers to leverage the lifetime of the [Enclave]
+    /// instance, preventing the call of SGX functions after the [Enclave] has
+    /// been dropped.
+    pub fn get_id(&self) -> Option<&sgx_enclave_id_t> {
+        Option::from(&self.id)
+    }
+
+    /// Create the enclave
+    ///
+    /// Will talk to the SGX SDK to create the enclave.  Once the enclave has
+    /// been created then calls on the enclave can be made.  See
+    /// [Enclave::get_id()]
+    ///
+    /// # Returns
+    ///
+    /// [_status_t_SGX_SUCCESS] when the enclave is created successfully.
+    ///
+    /// See
+    /// <https://download.01.org/intel-sgx/sgx-dcap/1.13/linux/docs/Intel_SGX_Enclave_Common_Loader_API_Reference.pdf>
+    /// for error codes and their meaning.
+    pub fn create(&mut self) -> sgx_status_t {
+        let mut launch_token: sgx_launch_token_t = [0; 1024];
+        let mut launch_token_updated: c_int = 0;
+        let mut misc_attr: sgx_misc_attribute_t =
+            unsafe { MaybeUninit::<sgx_misc_attribute_t>::zeroed().assume_init() };
+        let mut enclave_id: sgx_enclave_id_t = 0;
+        let result = unsafe {
+            sgx_create_enclave(
+                self.filename.as_ptr(),
+                self.debug as c_int,
+                &mut launch_token as *mut sgx_launch_token_t,
+                &mut launch_token_updated,
+                &mut enclave_id,
+                &mut misc_attr,
+            )
```

Comment:
As noted above, this should use [`sgx_create_enclave_from_buffer_ex`](https://download.01.org/intel-sgx/sgx-linux/2.16/docs/Intel_SGX_Developer_Reference_Linux_2.16_Open_Source.pdf#%5B%7B%22num%22%3A232%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C94.5%2C339%2C0%5D)

Reply:
Looking into this now, found this, https://github.com/intel/linux-sgx/issues/482 
Thinking we'll want to support both files and buffers, not sure how often we've needed to debug in the past, I know I did for the target triple spec file

Reply:
using `sgx_create_enclave_from_buffer_ex` is now utilized for the interface